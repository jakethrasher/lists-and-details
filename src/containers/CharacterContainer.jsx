import React, { Component } from 'react'
import CharacterList from '../components/characters/CharacterList';
import { fetchCharacters } from '../services/fetchCharacters';

export default class CharacterContainer extends Component {
  state={
    characters:[]
  }
  
  componentDidMount= async () => {
    const data = await fetchCharacters();
    this.setState({characters:data.body})
  }
  render() {
  
      return (
        <>
          <h1>Hey Arnold! Character List:</h1>
          <CharacterList characters={this.state.characters}/>
        </>
      )
    }
}
