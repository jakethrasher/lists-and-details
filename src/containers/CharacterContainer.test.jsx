import React from 'react';
import { render, screen } from '@testing-library/react';
import CharacterContainer from './CharacterContainer';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterContainer', () => {
  it('renders a list of characters from the hey arnold api', async () => {
    render(
      <MemoryRouter>
        <CharacterContainer/>
      </MemoryRouter>
    );

    screen.getByText('LOADING');

    const ul = await screen.findByRole('list', { name:'characters' });
    expect(ul).toMatchSnapshot();
        
  });
});
