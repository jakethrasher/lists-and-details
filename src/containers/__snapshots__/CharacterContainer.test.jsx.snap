// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer renders a list of characters from the hey arnold api 1`] = `
<ul
  aria-label="characters"
>
  <a
    href="/5da237699734fdcb7bef8f5c"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <p>
        Nancy
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f59"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <p>
        Grandpa's grandfather
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f52"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      />
      <p>
        Grandpa Phil
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f58"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      />
      <p>
        Grandpa's father
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f70"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
      />
      <p>
        Phoebe Heyerdahl
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f7e"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
      />
      <p>
        Eugene Horowitz
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f82"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
      />
      <p>
        Ray
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f8b"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
      />
      <p>
        Shelley
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f76"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
      />
      <p>
        Lila Sawyer
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f7c"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
      />
      <p>
        Uncle Stinky
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f55"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      />
      <p>
        Arnie
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f66"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
      />
      <p>
        Olga Pataki
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f6b"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
      />
      <p>
        Timberly Johanssen
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f6d"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
      />
      <p>
        Jerry Berman
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f73"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
      />
      <p>
        Rhonda Wellington Lloyd
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f78"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
      />
      <p>
        Stinky Peterson
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f71"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
      />
      <p>
        Kyo Heyerdahl
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f8e"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
      />
      <p>
        Iggy
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f77"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
      />
      <p>
        Mr. Sawyer
      </p>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f84"
  >
    <li>
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
      />
      <p>
        Thaddeus "Curly" Gammelthorpe
      </p>
    </li>
  </a>
</ul>
`;
